### PR TITLE
[8.0][IMP][base_location_nuts] Simplify, fix some bugs

### PR DESCRIPTION
**Pull request by yajo**
This makes NUTS labels and levels to be stored in the res.country object.
Now creating l10n submodules is a piece of cake!

Relational fields now follow guidelines on naming. Old name attribute used
for backwards compatibility wherever needed.

Also some methods have been renamed, and refactored to be smarter. Most cases
l10n modules will just need to fill the res.contry table, and regions and
substates domains will work out of the box. In case you still need to
overwrite any method, splitting in smaller methods makes it easier too.

Oh! And no need for recursive dictionary updates.

Some bugs in submodules get fixed on the way, when the opened PRs for them are updated (I'm on it, please wait https://github.com/OCA/l10n-germany/pull/11 and https://github.com/OCA/l10n-spain/pull/150).

@rafaelbn @antespi

### CODE REVIEW

**Comment by rafaelbn** [comment]
Please @JavierIniesta review, ping @antespi

**Comment by JavierIniesta** [comment]
LGTM :+1:

File: base_location_nuts/i18n/es.po [hunk 1]
```diff
@@ -190,7 +260,13 @@ msgid ""
 "                Updating or creating new NUTS entries if not\n"
 "                found already in the system, and DELETING MISSING\n"
 "                ENTRIES from new downloaded file."
-msgstr "Este asistente descargará la última version de\n                NUTS 2013 desde el servicio de matadatos europeo RAMON.\n                Actualizando o creando nuevas regiones NUTS si no\n                las encuentra en el sistemma, y BORRANDO LAS QUE NO ENCUENTRE\n                en el nuevo fichero descargado."
+msgstr ""
+"Este asistente descargará la última version de\n"
+"                NUTS 2013 desde el servicio de matadatos europeo RAMON.\n"
+"                Actualizando o creando nuevas regiones NUTS si no\n"
+"                las encuentra en el sistemma, y BORRANDO LAS QUE NO "
```
Comment (antespi) [inline base_location_nuts/i18n/es.po]: s/sistemma/sistema/

File: base_location_nuts/i18n/es.po [hunk 2]
```diff
@@ -190,7 +260,13 @@ msgid ""
 "                Updating or creating new NUTS entries if not\n"
 "                found already in the system, and DELETING MISSING\n"
 "                ENTRIES from new downloaded file."
-msgstr "Este asistente descargará la última version de\n                NUTS 2013 desde el servicio de matadatos europeo RAMON.\n                Actualizando o creando nuevas regiones NUTS si no\n                las encuentra en el sistemma, y BORRANDO LAS QUE NO ENCUENTRE\n                en el nuevo fichero descargado."
+msgstr ""
+"Este asistente descargará la última version de\n"
+"                NUTS 2013 desde el servicio de matadatos europeo RAMON.\n"
```
Comment (antespi) [inline base_location_nuts/i18n/es.po]: s/matadatos/metadatos/

File: base_location_nuts/README.rst [hunk 3]
```diff
@@ -44,6 +44,13 @@ l10n_de_location_nuts, ...) will inherit this wizard and
 relate each NUTS item with states. So if you install a new localization addon
 you must re-build NUTS clicking this wizard again.
 
+To configure lables and levels per country, you should install a l10n module,
```
Comment (antespi) [inline base_location_nuts/README.rst]: s/lables/labels/

File: base_location_nuts/README.rst [hunk 4]
```diff
@@ -56,7 +63,6 @@ in order to allow to assign them to partner object.
    :alt: Try me on Runbot
    :target: https://runbot.odoo-community.org/runbot/134/{branch}
```
Comment (antespi) [inline base_location_nuts/README.rst]: s/{branch}/8.0/

**Comment by antespi** [comment]
There's some issues in Travis lint_check, please @Yajo review my comments, thanks

File: base_location_nuts/models/res_partner.py [hunk 5]
```diff
@@ -5,57 +5,69 @@
...
+        fields.remove("state")
+        for field in fields:
+            level = self.country_id["%s_level" % field]
+            field += "_id"
+            if level:
+                domain[field].append(("level", "=", level))
+
+        return {
+            "domain": domain,
+        }
```
Comment (antespi) [inline base_location_nuts/models/res_partner.py:100]: Please, add nuts fields to `_address_fields` in order to be consider as partner address fields and then Odoo copies them when child partner has `use_parent_address` flag set:

```
    @api.model
    def _address_fields(self):
        fields = super(ResPartner, self)._address_fields()
        if fields:
            fields += ['substate_id', 'region_id']
        return fields
```

**Comment by antespi** [comment]
@Yajo you should fix these Travis issues: https://travis-ci.org/OCA/partner-contact/jobs/106487255#L406

**Comment by rafaelbn** [comment]
Thanks! Travis fixed, my test :+1:

**Comment by antespi** [comment]
:+1:

File: base_location_nuts/README.rst [hunk 6]
```diff
@@ -7,28 +7,22 @@ NUTS Regions
...
+* NUTS L4: Normally small region or province level
 
 
 Installation
 ============
 
-You need to install another addon (one for each country) in order to use
-these NUTS, for example:
+We recommend to install another addon (one for each country) in order to relate
+NUTS with states define by each localization addon, for example:
```
Comment (pedrobaeza) [inline base_location_nuts/README.rst]: s/define/defined

File: base_location_nuts/__openerp__.py [hunk 7]
```diff
@@ -24,16 +24,20 @@
 {
     'name': 'NUTS Regions',
     'category': 'Localisation/Europe',
-    'version': '8.0.1.0.0',
+    'version': '8.0.2.0.0',
```
Comment (pedrobaeza) [inline base_location_nuts/__openerp__.py:27]: There's no such big change, isn't it? I would put 8.0.1.1.0

**Comment by pedrobaeza** [comment]
Minor remarks, but :+1:

File: base_location_nuts/__openerp__.py (hunk 7 above)
Comment (yajo) [inline base_location_nuts/__openerp__.py:27]: They are breaking changes. Submodules must update or they will not work.

Comment (pedrobaeza) [inline base_location_nuts/__openerp__.py:27]: OK

**Pull request merged**